source/NVDAObjects/IAccessible/MSHTML.py

```diff
@@ -84,6 +85,23 @@ def __getitem__(self,item):
 		self.cache[item]=value
 		return value
 
+	def __contains__(self,item):
+		try:
+			return self.containsCache[item]
+		except LookupError:
+			pass
+		value = self[item]

```

Comment:
I would perhaps  only check item in self.containsCache, then item in self.cache, and if none of them then node.hasattribute(item) and add the answer to containsCache.  hasattribute should work for any value that getattribute would work for.